pkg/apis/build/v1alpha1/build_validation.go

```diff
@@ -92,3 +97,57 @@ func (bt *BuildSpec) validateTimeout() *apis.FieldError {
 	}
 	return nil
 }
+
+// Validate source
+func (bs BuildSpec) validateSources() *apis.FieldError {
+	var subPathExists bool
+	names := map[string]string{}
+	var sourceExists bool
```

Comment:
I don't think we gain much from making this a variable.

Line 112 could just be:

```
if len(bs.Sources) > 0 && b.Source != nil {
 // boom
}
```